dot/types/grandpa.go

```diff
@@ -193,3 +193,52 @@ type GrandpaVote struct {
 func (v *GrandpaVote) String() string {
 	return fmt.Sprintf("hash=%s number=%d", v.Hash, v.Number)
 }
+
+// GrandpaEquivocation is used to create a proof of equivocation
+type GrandpaEquivocation struct {
+	RoundNumber     uint64
+	ID              ed25519.PublicKey
+	FirstVote       GrandpaVote
+	FirstSignature  [64]byte
+	SecondVote      GrandpaVote
+	SecondSignature [64]byte
+}
+
+// GrandpaEquivocationVote is a custom vdt type for a grandpa equivocation
+type GrandpaEquivocationVote scale.VaryingDataType
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
```

Comment:
supernit `will` feels like it's doing async stuff, just use present?

```suggestion
// Set sets a VaryingDataTypeValue using the underlying VaryingDataType
```